Apps/phgooglepeople/googlepeople_connector.py

```diff
@@ -0,0 +1,570 @@
+                error_code = ERR_CODE_MSG
+                error_msg = ERR_MSG_UNAVAILABLE
+        except:
+            error_code = ERR_CODE_MSG
+            error_msg = ERR_MSG_UNAVAILABLE
+
+        try:
+            if error_code in ERR_CODE_MSG:
+                error_text = "Error Message: {0}".format(error_msg)
+            else:
+                error_text = "Error Code: {0}. Error Message: {1}".format(error_code, error_msg)
+        except:
+            self.debug_print(PARSE_ERR_MSG)
+            error_text = PARSE_ERR_MSG
+
+        return error_text
+
+    def _validate_integer(self, action_result, parameter, key):
+        if parameter is not None:
+            try:
+                if not float(parameter).is_integer():
+                    return action_result.set_status(phantom.APP_ERROR, INVALID_INTEGER_ERR_MSG.format(key)), None
+
+                parameter = int(parameter)
+            except:
+                return action_result.set_status(phantom.APP_ERROR, INVALID_INTEGER_ERR_MSG.format(key)), None
+
+            if parameter <= 0:
+                return action_result.set_status(phantom.APP_ERROR, INVALID_NON_ZERO_NON_NEGATIVE_INTEGER_ERR_MSG.format(key)), None
+
+        return phantom.APP_SUCCESS, parameter
+
+    def _create_client(self, action_result, scopes):
+        credentials = None
+        try:
+            credentials = service_account.Credentials.from_service_account_info(self._key_dict, scopes=scopes)
+        except Exception as e:
+            err_msg = self._get_error_message_from_exception(e)
+            return RetVal(action_result.set_status(phantom.APP_ERROR, "Unable to get the credentials from the key json. {}".format(err_msg)), None)
+
+        if self._login_email:
+            try:
+                credentials = credentials.with_subject(self._login_email)
+            except Exception as e:
+                err_msg = self._get_error_message_from_exception(e)
+                return RetVal(action_result.set_status(phantom.APP_ERROR, "Failed to create delegated credentials. {}".format(err_msg)), None)
+
+        try:
+            client = discovery.build('people', 'v1', credentials=credentials)
+        except Exception as e:
+            err_msg = self._get_error_message_from_exception(e)
+            return RetVal(action_result.set_status(phantom.APP_ERROR, "Unable to create client. {}".format(err_msg)), None)
+
+        return RetVal(phantom.APP_SUCCESS, client)
+
+    def _paginator(self, client, fields, limit):
+
+        """
+        This method repeatedly makes API calls until the requested number of records are fetched from the server.
+
+        :param client: The object of google client API
+        :param fields: The fields to be fetched
+        :param limit: The number of records to be fetched
+        """
+
+        kwargs = dict()
+        list_items = list()
+        page_token = None
+        action_id = self.get_action_identifier()
+        kwargs.update({'pageSize': 1000})
```

Comment:
Is this the default value suggested in the docs?

Reply:
This value is the maximum allowed value for the pageSize mentioned in the documentation. 
Link: https://developers.google.com/people/api/rest/v1/otherContacts/list

The default value is 100 but usually, if we implement pagination then we keep the maximum possible value in the page size limit. Hence, we have kept it as 1000.